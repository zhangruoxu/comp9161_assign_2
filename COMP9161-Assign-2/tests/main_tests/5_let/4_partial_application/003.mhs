main
     =
    let f =
        letfun f x = 
          letfun g y = x;
    in 
    let g = f 10;

    in g 3;

