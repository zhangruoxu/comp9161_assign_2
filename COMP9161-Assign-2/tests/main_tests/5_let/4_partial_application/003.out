main :: Int = let f :: forall a. forall c. (a -> c -> a) = (letfun f :: (a -> c -> a) x = (letfun g :: (c -> a) y = x));
              in let g :: forall f. (f -> Int) = f 10;
                 in g 3;
